core/config.go

```diff
@@ -1,12 +1,54 @@
 package autospotting
 
 import (
+	"fmt"
 	"io"
+	"strings"
 	"time"
 
 	"github.com/cristim/ec2-instances-info"
 )
 
+// String formats the ArrayOfStrings type string[]
+// into a string
+func (i *Tags) String() string {
+	if len(*i) > 0 {
```

Comment:
I don't think this will ever be null, as the flags parsing is making it not.  All the examples I've seen, not one is checking for nil (see next comment for references),

And when running autospotting on command line, without the -tag_filters, it's not segmentation faulting

but I'll add it to be on the same side, if needed.

Reply:
I know and I fully agree with you, there are indeed various pointers/slice pointers which aren't expected to be nil. I simply consider it a best practice to check anyway, as a basic reflex to avoid such case especially because it has little to no-cost performance wise.